daemon/appinfo.c

```diff
@@ -213,27 +231,28 @@ static const char appinfo_unknown[] = "(undefined)";
 static void
 appinfo_ctor(appinfo_t *self, applications_t *applications, const gchar *id)
 {
-    self->anf_applications         = applications;
-    self->anf_appname              = g_strdup(id);
+    self->anf_applications               = applications;
+    self->anf_appname                    = g_strdup(id);
 
-    self->anf_state                = APPINFO_STATE_UNSET;
-    self->anf_dt_ctime             = -1;
-    self->anf_dirty                = false;
+    self->anf_state                      = APPINFO_STATE_UNSET;
+    self->anf_dt_ctime[APPINFO_DIR_MAIN] = -1;
+    self->anf_dt_ctime[APPINFO_DIR_ALT]  = -1;
```

Comment:
Hmm, yeah, I can change it to either direction. Although, I did some searching whether gcc does loop unrolling or not and it seems to be so that our optimization level doesn't do loop unrolling or peeling. We would need to up that from -O2 to -O3 or explicitly ask for it.

https://gcc.gnu.org/onlinedocs/gcc-8.5.0/gcc/Optimize-Options.html#Optimize-Options

Reply:
Removed the loop from application_t constructor instead.